### PR TITLE
Initial LogindConsoleServices implementation

### PR DESCRIPTION
**Pull request by RAOF**
The astute reader will see that this is not actually hooked up to anything yet. That's because it's incomplete.

However, there's probably enough here to be sensibly reviewed, and that should make further implementation easier.

### CODE REVIEW

**Comment by RAOF** [comment]
And, indeed, I get to find out that `python3` is broken on 16.04 arm64, that Fedora has an old `gdbus-codegen`, and that `libglib2.0-dev-bin` doesn't exist in 16.04. Bah!

**Comment by Conan-Kudo** [comment]
@RAOF Uhh, how do we have an old `gdbus-codegen` in Fedora? Fedora 27 has glib2-2.54.3, and Fedora 28 and Rawhide have glib2-2.56.0, which is the newest version at this time?

**Comment by RAOF** [comment]
@Conan-Kudo I'm unsure what was happening with Fedora, but the `gdbus-codegen` binary that the builds used complained about not understanding the `--body` or `--header` options, which are the non-deprecated way of generating code.

**Comment by RAOF** [comment]
Heh. It seems Fedora's `python3-dbusmock` package is missing a dependency on `python3-gi` :grin:

File: src/server/console/logind_console_services.cpp [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+#include "mir/fd.h"
+#include "mir/main_loop.h"
+
+#define MIR_LOG_COMPONTENT "logind"
+#include "mir/log.h"
+
+namespace
+{
+class GErrorPtr
```
Comment (AlanGriffiths) [inline src/server/console/logind_console_services.cpp]: I think GErrorPtr Copy/Assign should be  "= delete"

**Pull request merged**